wapitiCore/net/jsparser_angular.py

```diff
@@ -0,0 +1,81 @@
+import enum
+import re
+from urllib.parse import urlparse
+
+class JsParserAngular:
+    """
+    A class that try to parse JS files from angular webapps
+    and extract interesting subdomains and paths.
+    """
+
+    def __init__(self, url: str, data: str):
+        """
+        Create a new JsParserAngular object.
+
+        Takes the following arguments:
+            url: The absolute URL of the JS file on the server
+            data: The content of the JS file as a string
+        """
+        self.links = []
+
+        parts = urlparse(url)
+        netloc = parts.netloc
+        self.target_domain = netloc
+
+        if netloc and len(netloc.split('.')) > 2:
+            netloc = '.'.join(netloc.split('.')[1:])
+
+        self.domain = netloc
+        self.scheme = parts.scheme
+
+        if self.domain:
```

Comment:
Is there any possible case where `self.domain` is not set ?

Reply:
Only if we have an invalid url